sql/src/main/java/org/apache/druid/sql/calcite/planner/DruidRexExecutor.java

```diff
@@ -176,6 +176,17 @@ public void reduce(
               }
               literal = rexBuilder.makeLiteral(resultAsBigDecimalList, constExp.getType(), true);
             }
+          } else if (constExp.getType().getComponentType().getSqlTypeName() == SqlTypeName.BOOLEAN) {
+            List<Boolean> resultAsBooleanList = new ArrayList<>(array.length);
+            for (Object val : exprResult.castTo(ExpressionType.LONG_ARRAY).asArray()) {
+              final Number longVal = (Number) val;
+              if (longVal == null) {
+                resultAsBooleanList.add(null);
+              } else {
+                resultAsBooleanList.add(longVal.longValue() > 0);
```

Comment:
nit: could use `Evals.asBoolean` here just on the off chance we ever change how we coerce longs into booleans in the native layer, but this is probably safe too